src/libsyntax/ext/deriving/show.rs

```diff
@@ -72,70 +67,53 @@ fn show_substructure(cx: &mut ExtCtxt, span: Span,
         }
     };
 
-    let mut format_string = String::from_str(&token::get_ident(name));
-    // the internal fields we're actually formatting
-    let mut exprs = Vec::new();
+    // We want to make sure we have the expn_id set so that we can use unstable methods
+    let span = Span { expn_id: cx.backtrace(), .. span };
```

Comment:
Did you find that this was necessary? I'm under the impression that the macro infrastructure manages re-spanning everything after a macro expansion?